docs/cow-protocol/reference/contracts/periphery/cow_uid_generator.md

```diff
@@ -0,0 +1,66 @@
+---
+id: cow-uid-generator
+sidebar_position: 4
+---
+
+# CoWUidGenerator
+
+A helper contract for calculating the same EIP-712 signature hash for a given user order that the GPv2SettlementContract expects.
+
+## Architecture
+This contract is a simple helper contract that:
+
+- References the CoW Protocol settlement contract’s `domainSeparator`.
+- Constructs a Gnosis Protocol v2 (CoW) order data struct on-chain.
+- Determines whether the order is a sell or buy order based on `isSell`.
+- Computes the EIP-712 hash that matches what the CoW Protocol uses to verify user signatures off-chain.
+
+## Data Types and Storage
+
+### `GPv2SettlementContract`
+An interface for the [CoW Protocol settlement contract](../core/settlement.md).
+
+```solidity
+interface GPv2SettlementContract {
+    function domainSeparator() external view returns (bytes32);
+}
+```
+
+## Functions
+
+### `getUid`
+Constructs an order, determines if it is a sell or buy order, then computes the same EIP-712 hash used by CoW Protocol for verifying signatures. Returns both the EIP-712 digest and the ABI-encoded order.
```

Comment:
please add a note here that the thing that is being returned here is just the order digest. In order to get the order uid it needs to be concatenated with the trader and valid from (this is a common gotch)

![image](https://github.com/user-attachments/assets/b9e691d9-ea83-4db3-9690-106e83e8c936)